components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -2218,13 +2206,23 @@ private Set<AccessTokenDO> getAccessTokensByTenant(int tenantId, String userStor
                     String authzUser = resultSet.getString(10);
                     userStoreDomain = resultSet.getString(11);
                     String consumerKey = resultSet.getString(12);
+                    String authorizedOrganization = resultSet.getString(13);
                     String authenticatedIDP = null;
                     if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                        authenticatedIDP = resultSet.getString(13);
+                        authenticatedIDP = resultSet.getString(14);
                     }
 
                     AuthenticatedUser user = OAuth2Util.createAuthenticatedUser(authzUser, userStoreDomain,
                             OAuth2Util.getTenantDomain(tenantId), authenticatedIDP);
+                    /* For organization bound access tokens, the authenticated user should be populated considering
```

Comment:
Move this logic into the the create Authenticated user method.

Reply:
There are multiple `createAuthenticatedUser` implementations in the `OAuth2Util` https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L4055, hence it is bit tricky to move the logic to that method.

We can further revisit the implementation of fetching app tenant ID and improve the code appropriately.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2504/commits/7771cbb1435e4123f6e0af7d0b64327845acbb85